framework/common/encryption.cpp

```diff
@@ -223,61 +223,112 @@ bool hmac::digest(uint8_t *digest)
     return HMAC_Final(m_ctx, digest, &digest_length);
 }
 
-bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *data, uint32_t data_len)
+bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *plaintext, int plen,
+                 uint8_t *ciphertext, int &clen)
 {
     EVP_CIPHER_CTX *ctx;
+    int len;
 
-    int clen, len;
-    uint8_t buf[128];
-
+    /* Create and initialise the context */
     ctx = EVP_CIPHER_CTX_new();
     if (NULL == ctx) {
+        MAPF_ERR("Failed to create context");
         return false;
     }
+
+    /*
+     * Initialise the encryption operation.
+     * IMPORTANT - key and IV should be
+     * The same as block size, which 16 bytes since we are using 
+     * 128 bit AES (i.e. a 128 bit key). The IV size for *most* modes is
+     * the same as the block size.
```

Comment:
This kind of comment would be a lot more relevant in the doxygen in the header file, since it's about the interface, not the implementation.